routes/libraries.js

```diff
@@ -274,45 +111,273 @@ router
+     *  genres: [array<string>]
+     * }
+     * This is using the error checking from before, using the helper functions for uniformity.
+     */
+    try {
+      updateData.fullness = validation.isValidNumber(
+        updateData.fullness,
+        "Fullness Rating"
+      );
+      updateData.genres = validation.checkStringArray(
+        updateData.genres,
+        "Genres Available"
+      );
+    } catch (e) {
+      return res.status(400).json({ error: e });
+    }
+    // At this point, assume the form data is completely valid.
+    try {
+      const { fullness, genres } = updateData;
+      const updatedLibrary = await libraryData.formUpdate(
+        libraryId,
+        fullness,
+        genres
+      );
+      res.json(updatedLibrary);
+    } catch (e) {
+      res.status(500).render('error', {errorCode: 500, title: "Error Page"});
+    }
+  });
+
+router.route('/:id/comments')
+  .get(async (req, res) => {
+    // Renders the comments section
+  })
   .post(async (req, res) => {
-    //GET USER ID FROM MIDDLEWARE COOKIE
-    let id = req.params.id;
+    // Creates a new comment
+    let id;
+    
+    // If the library ID is not valid, render the error page with a status code of 400
+    try {
+      id = req.params.id;
+      id = validation.checkValidId(id);
+    } catch (e) {
+      res.status(400).render('error', {errorCode: 400, searchValue: "Library"});
+    }
+
+    let text;
+    
+    try {
+      text = req.body.text;
+      text = validation.checkString(text);
+    } catch (e) {
+      res.status(400).render('libraries/library', {errorCode: 400, searchValue: "Library"}); // RENDER ERROR MESSAGE ON CREATE COMMENTS SECTION
+    }
     
-    id = validation.checkValidId(id);
+    let library;
+
+    // If the library is not found, render the error page with a status code of 404
+    try {
+      library = await libraryData.get(id);
+    } catch (e) {
+      return res.status(404).render('error', {errorCode: 404, searchValue: "Library"});
+    }
 
-    let commentId = req.body.addCommentLike;
+    try {
+      let userid = req.session.user._id.toString();
+      let createComment = await libraryData.createComment(id, userid, text);
+      res.redirect(`libraries/library/${id}`);
+    } catch (e) {
+      res.status(500).render('error', {errorCode: 500, title: "Error Page"});
+    }
+
+    // try {
+    //   newLibraryData.ownerID = validation.checkValidId(
+    //     newLibraryData.ownerID,
+    //     "Library Owner ID"
+    //   );
+    // } catch (e) {
+    //   errors.push(e);
+    // }
+    // try {
+    //   // TODO: THIS WILL BE UPDATED BECAUSE THE WAY OF SERVY CHANGING
+    //   newLibraryData.fullness = parseInt(newLibraryData.fullness);
+    //   newLibraryData.fullness = validation.isValidNumber(
+    //     newLibraryData.fullness,
+    //     "Fullness Rating"
+    //   );
+    //   if (0 > newLibraryData.fullness || newLibraryData.fullness > 5) {
+    //     throw "Fullness rating must be between 0-5";
```

Comment:
Is this commented out code for future or we don't want this code anymore ? If we don't want it anymore just remove it

Reply:
We might want it, but it was somehow misplaced here.